src/main/java/com/denizenscript/depenizen/bukkit/objects/breweryx/BreweryRecipeTag.java

```diff
@@ -0,0 +1,367 @@
+        }
+
+        BRecipe recipe = BRecipe.get(string);
+        if (recipe == null) {
+            return null;
+        }
+        return new BreweryRecipeTag(recipe);
+    }
+
+    public static boolean matches(String arg) {
+        arg = arg.replace("breweryrecipe@", "");
+        return BRecipe.get(arg) != null;
+    }
+
+    BRecipe bRecipe;
+
+    public BreweryRecipeTag(BRecipe bRecipe) {
+        this.bRecipe = bRecipe;
+    }
+
+    String prefix = "breweryrecipe";
+
+    @Override
+    public String getPrefix() {
+        return prefix;
+    }
+
+    @Override
+    public boolean isUnique() {
+        return true;
+    }
+
+    @Override
+    public String identify() {
+        return "breweryrecipe@" + bRecipe.getId();
+    }
+
+    @Override
+    public String identifySimple() {
+        return identify();
+    }
+
+    @Override
+    public ObjectTag setPrefix(String aString) {
+        prefix = aString;
+        return this;
+    }
+
+    public static void register() {
+        // <--[tag]
+        // @attribute <BreweryRecipeTag.id>
+        // @returns ElementTag
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the ID of the recipe.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "id", (attribute, object) -> {
+            String id = object.bRecipe.getId();
+            return new ElementTag(id, true);
+        });
+
+        // <--[tag]
+        // @attribute <BreweryRecipeTag.name>[<quality>]
+        // @returns ElementTag
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the name of the recipe based on the provided quality number.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, ElementTag.class, "name", (attribute, object, quality) -> {
+            return new ElementTag(object.bRecipe.getName(quality.asInt()), true);
+        });
+
+        // <--[tag]
+        // @attribute <BreweryRecipeTag.full_name>
+        // @returns ElementTag
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the full name of the recipe as declared in the recipes file.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "full_name", (attribute, object) -> {
+            return new ElementTag(String.join("/", object.bRecipe.getName()), true);
+        });
+
+        // <--[tag]
+        // @attribute <BreweryRecipeTag.ingredients>
+        // @returns ListTag(ItemTag)
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns a ListTag of ItemTags that are the ingredients of the recipe.
+        // -->
+        tagProcessor.registerTag(ListTag.class, "ingredients", (attribute, object) -> {
+            ListTag ingredients = new ListTag();
+            for (RecipeItem recipeItem : object.bRecipe.getIngredients()) {
+                ingredients.add(recipeItem.toConfigString());
+            }
+            return ingredients;
```

Comment:
Can use the `ListTag` convertor constructor (pass in the list, and a function that converts `RecipeItem`s to a Denizen object as a lambda)